plaso/parsers/text_plugins/macos_launchd.py

```diff
@@ -0,0 +1,170 @@
+    self.severity = None
+    self.written_time = None
+
+
+class MacOSLaunchdLogTextPlugin(interface.TextPlugin):
+  """Text parser plugin for MacOS launchd log files."""
+
+  NAME = 'macos_launchd_log'
+  DATA_FORMAT = 'MacOS launchd log file'
+
+  _TWO_DIGITS = pyparsing.Word(pyparsing.nums, exact=2).set_parse_action(
+      lambda tokens: int(tokens[0], 10))
+
+  _FOUR_DIGITS = pyparsing.Word(pyparsing.nums, exact=4).set_parse_action(
+      lambda tokens: int(tokens[0], 10))
+
+  _SIX_DIGITS = pyparsing.Word(pyparsing.nums, exact=6).set_parse_action(
+      lambda tokens: int(tokens[0], 10))
+
+  _DATE_TIME = (
+      _FOUR_DIGITS + pyparsing.Suppress('-') +
+      _TWO_DIGITS + pyparsing.Suppress('-') +
+      _TWO_DIGITS +
+      _TWO_DIGITS + pyparsing.Suppress(':') +
+      _TWO_DIGITS + pyparsing.Suppress(':') +
+      _TWO_DIGITS + pyparsing.Suppress('.') +
+      _SIX_DIGITS)
+
+  _PROCESS_NAME = (
+      pyparsing.Suppress('(') +
+      pyparsing.OneOrMore(
+          pyparsing.Word(pyparsing.printables, excludeChars=')'), stop_on=')') +
+      pyparsing.Suppress(')')).set_parse_action(' '.join)
+
+  _SEVERITY = pyparsing.Combine(
+      pyparsing.Suppress('<') +
+      pyparsing.Word(pyparsing.printables, excludeChars='>') +
+      pyparsing.Suppress('>'))
+
+  _LOG_LINE = (
+    _DATE_TIME.set_results_name('date_time') +
+    pyparsing.Optional(_PROCESS_NAME.set_results_name('process_name')) +
+    _SEVERITY.set_results_name('severity') +
+    pyparsing.Suppress(': ') +
+
+    pyparsing.restOfLine().set_results_name('body') +
+    pyparsing.Suppress(pyparsing.LineEnd()))
+
+  _LINE_STRUCTURES = [('log_line', _LOG_LINE)]
+
+  VERIFICATION_GRAMMAR = _LOG_LINE
+
+  def __init__(self):
+    """Initializes a text parser plugin."""
+    super(MacOSLaunchdLogTextPlugin, self).__init__()
+    self._event_data = None
+
+  def _ParseFinalize(self, parser_mediator):
+    """Finalizes parsing.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+    """
+    if self._event_data:
+      parser_mediator.ProduceEventData(self._event_data)
+      self._event_data = None
+
+  def _ParseLogline(self, structure):
+    """Parses a log line.
+
+    Args:
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+    """
+    time_elements_structure = self._GetValueFromStructure(
+        structure, 'date_time')
+
+    event_data = MacOSLaunchdEventData()
+    event_data.body = self._GetValueFromStructure(structure, 'body')
+    event_data.process_name = self._GetValueFromStructure(
+        structure, 'process_name')
+    event_data.severity = self._GetValueFromStructure(structure, 'severity')
+    event_data.written_time = (
+        dfdatetime_time_elements.TimeElementsInMicroseconds(
+            time_elements_tuple=time_elements_structure))
+
+    self._event_data = event_data
```

Comment:
@rick-slin why keep `event_data` around and not "produce" it `parser_mediator.ProduceEventData()` ? Are you planning to make changes to extend event_data e.g. for multi line log format /